processor/processor.go

```diff
@@ -797,6 +798,106 @@ func (proc *HandleT) updateSourceEventStatsDetailed(event types.SingularEventT,
 		}
 	}
 }
+
+type TransformationStatusT struct {
+	sourceID          string
+	destID            string
+	destination       *backendconfig.DestinationT
+	eventsToTransform []transformer.TransformerEventT
+	eventsByMessageID map[string]types.SingularEventWithReceivedAt
+	response          *transformer.ResponseT
```

Comment:
only include failedEvents from response in this struct since we are only using failedEvents from response?